src/app/components/wrapper-components/part-creation/part-creation.component.spec.ts

```diff
@@ -9,620 +12,715 @@ import { PartDAO } from 'src/app/dao/PartDAO';
+                Utils.getNonNullable(component.configFormGroup.get('totalPartDuration')).setValue(1000);
 
                 // When a candidate arrives and is proposed a config
-                await mockCandidateArrival();
-                await chooseOpponent();
-                spyOn(joinerDAOMock, 'update').and.callThrough();
-                await component.proposeConfig();
+                mockCandidateArrival();
+                chooseOpponent();
+                spyOn(joinerDAO, 'update').and.callThrough();
+                proposeConfig();
 
                 // Then the data sent should be what creator saw
-                expect(joinerDAOMock.update).toHaveBeenCalledOnceWith('joinerId', {
+                expect(joinerDAO.update).toHaveBeenCalledOnceWith('joinerId', {
                     partStatus: PartStatus.CONFIG_PROPOSED.value,
                     partType: PartType.CUSTOM.value,
                     maximalMoveDuration: 100,
                     totalPartDuration: 1000,
-                    chosenPlayer: 'firstCandidate',
+                    chosenPlayer: UserMocks.OPPONENT_MINIMAL_USER,
                     firstPlayer: FirstPlayer.RANDOM.value,
                 });
+                component.stopSendingPresenceTokensAndObservingUsersIfNeeded();
             }));
-            it('should support blitz part', fakeAsync(async() => {
+            it('should support blitz part', fakeAsync(() => {
                 // Given a component with a candidate where blitz is selected
-                testUtils.detectChanges();
-                await mockCandidateArrival();
-                await chooseOpponent();
-                await testUtils.clickElement('#partTypeBlitz');
+                awaitComponentInitialisation();
+                mockCandidateArrival();
+                chooseOpponent();
+                clickElement('#partTypeBlitz');
 
-                spyOn(joinerDAOMock, 'update').and.callThrough();
+                spyOn(joinerDAO, 'update').and.callThrough();
 
                 // When proposing the config
-                await component.proposeConfig();
-                testUtils.detectChanges();
+                proposeConfig();
 
                 // The blitz should be part of it
-                expect(joinerDAOMock.update).toHaveBeenCalledOnceWith('joinerId', {
+                expect(joinerDAO.update).toHaveBeenCalledOnceWith('joinerId', {
                     partStatus: PartStatus.CONFIG_PROPOSED.value,
                     partType: PartType.BLITZ.value,
                     maximalMoveDuration: 30,
                     totalPartDuration: 900,
-                    chosenPlayer: 'firstCandidate',
+                    chosenPlayer: UserMocks.OPPONENT_MINIMAL_USER,
                     firstPlayer: FirstPlayer.RANDOM.value,
                 });
+                component.stopSendingPresenceTokensAndObservingUsersIfNeeded();
```

Comment:
tiens pas moyen de faire { ...InitialJoinerOrSo, partType: BLITZ } plutôt ?

Reply:
Le code était pas comme ça avant et tu m'avais fait changer ou j'ai rêvé ?
Sinon dans le "to have been called" on peut pas mettre qqch qui représente l'état.
Cependant avoir des champs qui représentent les changements standards peut être pas mal pour voir le cycle de vie.

Tiens, tu connais miro ? C'est un truc pour faire des graphes et organigramme, p-e que ça pourrait être sympa d'en faire deux trois pour représenter l'archi MGP non ?

Reply:
hmm non au contraire, j'ai déjà demandé de mettre en commun les parties communes, ce que je repropose ici. Sinon je comprends pas le reste de ta réponse ?

Miro : oui, on peut tout à fait faire ça, mais à condition d'utiliser un outil libre alors, vu qu'il y a des centaines d'outils pour ça ! Inkscape ou https://app.diagrams.net/ par exemple

Reply:
Ici je teste un "toHaveBeenCalledWith(change)" et pas "toHaveEndedWithState(stateAfterChange)" hors je peux pas diminuer change en "stateAfterChange", la simplification serait aller de stateAfterChange à "...state, change", je sais pas si j'suis plus clair là :P ?

Oui va pour le libre en espérant qu'y ai du pratique !
Ticket crée :p